sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -410,6 +448,86 @@ void analyseBatchSentimentShowStatsRunner(
+        final TextAnalyticsClientBuilder clientBuilder = createClientBuilder(getEndpoint(),
+            new TextAnalyticsApiKeyCredential(getApiKey()));
+        testRunner.apply(clientBuilder).accept(DETECT_LANGUAGE_INPUTS.get(0), DETECTED_LANGUAGE_ENGLISH);
+    }
+
+    void clientBuilderWithInvalidApiKeyCredentialRunner(
+        Function<TextAnalyticsClientBuilder, BiConsumer<String, HttpResponseException>> testRunner) {
+        final TextAnalyticsClientBuilder clientBuilder = createClientBuilder(getEndpoint(),
+            new TextAnalyticsApiKeyCredential(INVALID_KEY));
+        testRunner.apply(clientBuilder).accept(DETECT_LANGUAGE_INPUTS.get(0), HTTP_RESPONSE_EXCEPTION_CLASS);
+    }
+
+    void clientBuilderWithRotateToInvalidKeyRunner(
+        Function<TextAnalyticsClientBuilder, BiConsumer<String, HttpResponseException>> testRunner) {
+        final TextAnalyticsApiKeyCredential credential = new TextAnalyticsApiKeyCredential(getApiKey());
+        final TextAnalyticsClientBuilder clientBuilder = createClientBuilder(getEndpoint(), credential);
+        // Update to invalid key
+        credential.updateCredential(INVALID_KEY);
+        testRunner.apply(clientBuilder).accept(DETECT_LANGUAGE_INPUTS.get(0), HTTP_RESPONSE_EXCEPTION_CLASS);
+    }
+
+    void clientBuilderWithRotateToValidKeyRunner(
+        Function<TextAnalyticsClientBuilder, BiConsumer<String, DetectedLanguage>> testRunner) {
+        final TextAnalyticsApiKeyCredential credential = new TextAnalyticsApiKeyCredential(INVALID_KEY);
+        final TextAnalyticsClientBuilder clientBuilder = createClientBuilder(getEndpoint(), credential);
+        // Update to valid key
+        credential.updateCredential(getApiKey());
+        testRunner.apply(clientBuilder).accept(DETECT_LANGUAGE_INPUTS.get(0), DETECTED_LANGUAGE_ENGLISH);
+    }
+
+    void clientBuilderWithNullServiceVersionRunner(
+        Function<TextAnalyticsClientBuilder, BiConsumer<String, DetectedLanguage>> testRunner) {
+        final TextAnalyticsClientBuilder clientBuilder =
+            createClientBuilder(getEndpoint(), new TextAnalyticsApiKeyCredential(getApiKey()))
+            .retryPolicy(new RetryPolicy())
+            .serviceVersion(null);
+        testRunner.apply(clientBuilder).accept(DETECT_LANGUAGE_INPUTS.get(0), DETECTED_LANGUAGE_ENGLISH);
+    }
+
+    void clientBuilderWithDefaultPipelineRunner(
+        Function<TextAnalyticsClientBuilder, BiConsumer<String, DetectedLanguage>> testRunner) {
+        final TextAnalyticsClientBuilder clientBuilder =
+            createClientBuilder(getEndpoint(), new TextAnalyticsApiKeyCredential(getApiKey()))
+            .configuration(Configuration.getGlobalConfiguration())
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));
+        testRunner.apply(clientBuilder).accept(DETECT_LANGUAGE_INPUTS.get(0), DETECTED_LANGUAGE_ENGLISH);
+    }
+
+    void clientBuilderWithDefaultCountryHintRunner(
```

Comment:
It might be interesting to add a test where there is a default country hint. Then the user sets a different value for country hint in the request. 
The later should take precedent.